apps/src/main.rs

```diff
@@ -109,27 +109,18 @@ async fn main() -> Result<()> {
 
     // Encode the input and upload it to the storage provider.
     tracing::info!("Number to publish: {}", args.number);
-    let input = InputBuilder::new()
-        .write_slice(&U256::from(args.number).abi_encode())
-        .build();
+    let input_builder = InputBuilder::new().write_slice(&U256::from(args.number).abi_encode());
+    tracing::info!("input builder: {:?}", input_builder);
 
-    // If the input exceeds 2 kB, upload the input and provide its URL instead, as a rule of thumb.
-    let request_input = if input.len() > 2 << 10 {
-        let input_url = boundless_client.upload_input(&input).await?;
-        tracing::info!("Uploaded input to {}", input_url);
-        Input::url(input_url)
-    } else {
-        tracing::info!("Sending input inline with request");
-        Input::inline(input.clone())
-    };
+    let input_guest = input_builder.clone().build_env().unwrap();
+    let input_guest_bytes = input_guest.encode().unwrap();
```

Comment:
Maybe better to drop those unwraps, and use `?`(or `anyhow::context`).
Very opinionated, we could rename `input_guest` to `guest_env`